rust/lance-index/src/scalar/bitmap.rs

```diff
@@ -314,45 +316,76 @@ fn get_batch_from_arrays(
     Ok(RecordBatch::try_new(schema, columns)?)
 }
 
-// Takes an iterator of Vec<u64> and processes each vector
-// to turn it into a RoaringTreemap. Each RoaringTreeMap is
-// serialized to bytes. The entire collection is converted to a BinaryArray
-fn get_bitmaps_from_iter<I>(iter: I) -> Arc<dyn Array>
-where
-    I: Iterator<Item = RowIdTreeMap>,
-{
+async fn write_bitmap_index(
+    state: HashMap<ScalarValue, RowIdTreeMap>,
+    index_store: &dyn IndexStore,
+) -> Result<()> {
+    if state.is_empty() {
```

Comment:
the previous code has
```
    let keys_array = if state.is_empty() {
        new_empty_array(value_type)
    } else {
        ScalarValue::iter_to_array(keys_iter)?
    };
```

i wonder do we need to write anything if state is empty?